### PR TITLE
Refactor for non-ube APR

### PR DESCRIPTION
**Pull request by randyn**
Addresses https://github.com/Ubeswap/ubeswap-interface/issues/252 to provide APR calculations even when the reward token isn't ube

A few different things in this PR, I'm cool with breaking it up or cutting them out and creating a new PR if we either need to table a refactor or decide to move in a different direction:

1. Moved annual reward calculation to a separate hook (`useAnnualReward`) and called it from `PoolCard`. I did this because calling `useCUSD` on a list of staking pools didn't work so well. My moving it to the top level of a React component, we could have a reliable number of hook calls and that made React a bit happier
2. Moved `useStakingInfo` hook to its own file. I mostly did this because the stake hooks file was getting pretty big and it was making it hard to navigate to the code I wanted to look at when it was packaged with the other staking related hooks. Open to moving it back if we feel like we want to stick with the previous pattern
3. Created `PoolStatRow` to simplify `PoolCard` a tiny bit

### CODE REVIEW

File: src/components/earn/PoolCard.tsx [hunk 1]
```diff
@@ -78,6 +81,8 @@ export const PoolCard: React.FC<Props> = ({ stakingInfo, dualRewards }: Props) =
   // get the color of the token
   const token = token0.symbol?.startsWith('m') ? token1 : token0
   const backgroundColor = useColor(token)
+  const { chainId } = useActiveWeb3React()
```
Comment (BriungRi) [inline src/components/earn/PoolCard.tsx:84]: Change to `useContractKit`

File: src/components/earn/PoolStats/PoolStatRow.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import React from 'react'
+import { TYPE } from 'theme'
+
+export interface PoolRewardProps {
+  helperText?: React.ReactNode
+  statName: string
+  statValue?: string
+}
+
+export default function PoolStatRow({ helperText, statName, statValue }: PoolRewardProps) {
```
Comment (BriungRi) [inline src/components/earn/PoolStats/PoolStatRow.tsx:12]: nice

**Comment by BriungRi** [comment]
@randyn Awesome work and sorry for my review delay (will be better about this in the future).

Two things:
 - We're no longer using `useActiveWeb3`, so make sure to fetch things with `useContractKit`
 - There are now merge conflicts because of a wallet connection update I made on the `main` branch.

If you could just address those, I'll merge your changes in.

**Comment by randyn** [comment]
> @randyn Awesome work and sorry for my review delay (will be better about this in the future).
> 
> Two things:
> 
> * We're no longer using `useActiveWeb3`, so make sure to fetch things with `useContractKit`
> * There are now merge conflicts because of a wallet connection update I made on the `main` branch.
> 
> If you could just address those, I'll merge your changes in.

Updated!

**Comment by BriungRi** [comment]
Looks great! Thanks again <3

**Comment by goldencircle777** [comment]
Thanks Randy ! Checkout your address, we have a gift for you

**Pull request merged**